auslib/web/admin/swagger/api.yaml

```diff
@@ -36,129 +50,418 @@ tags:
+      consumes: []
+      produces:
+        - application/json
+      externalDocs:
+        url: "http://mozilla-balrog.readthedocs.io/en/latest/database.html#user-roles"
+        description: Returns all distinct user roles
+      responses:
+        '200':
+          description: Successfully fetched all user roles.
+          schema:
+            type: object
+            properties:
+              roles:
+                description: list of roles
+                type: array
+                uniqueItems: true
+                items:
+                  type: string
+                  minItems: 0
+                example: ["releng", "qa"]
+
+  /users/{username}:
+    get:
+      summary: Returns all of the details about the user
+      description: >
+        Returns all of the details about the user's roles and permissions [Docs](http://mozilla-balrog.readthedocs.io/en/latest/database.html#permissions).
+      tags:
+        - Users
+      operationId: auslib.web.admin.views.mapper.specific_user_get
+      consumes: []
+      produces:
+        - application/json
+      externalDocs:
+        url: "http://mozilla-balrog.readthedocs.io/en/latest/database.html#user-roles"
+        description: Returns all of the details about the user
+      parameters:
+        - $ref: '#/parameters/usernameParam'
+      responses:
+        '200':
+          description: Successfully fetched all user details.
+          schema:
+            type: object
+            required:
+              - username
+              - permissions
+              - roles
+            properties:
+              username:
+                description: unique identifier name of the user
+                type: string
+                maxLength: 100
+                minLength: 1
+                example: balrogadmin
+              roles:
+                description: "user's assigned roles"
+                type: object
+                # Cannot fully validate roles object as patternProperties, part of jsonschema, isn't supported by
+                # swagger. https://github.com/OAI/OpenAPI-Specification/issues/687
+                example:
+                  sample_role:
+                    data_version: 1
+              permissions:
+                $ref: "#/definitions/UserPermissionsGET"
+          examples:
+            application/json:
+              permissions:
+                rule:
+                  data_version: 1
+                  options:
+                    actions:
+                      - create
+                      - modify
+              roles:
+                temp:
+                  data_version: 1
+              username: balrogadmin
+        '404':
+          description: No permission found for given username
+          schema:
+            $ref: "#/definitions/4xxClientErrorResponseModel"
+          examples:
+            application/json:
+              detail: "No permission found for username"
+              status: 404
+        '403':
```

Comment:
Are you sure this is a valid response? As far as I know we have no restrictions around who can view permissions.